hazelcast/src/main/java/com/hazelcast/map/impl/recordstore/MetadataStore.java

```diff
@@ -17,32 +17,83 @@
 package com.hazelcast.map.impl.recordstore;
 
 import com.hazelcast.internal.serialization.Data;
+import com.hazelcast.query.impl.AbstractMetadata;
 import com.hazelcast.query.impl.Metadata;
 
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
 
-public class MetadataStore {
+/**
+ * On-heap Metadata Store. Uses CHM to store the key/metadata pairs.
```

Comment:
Maybe you can also document the reason of CHM usage instead of not thread safe HashMap, this can be helpful.

Reply:
I actually took this implementation from the existing MetadataStore implementation. It looks like it is called only from the partition thread and we can use HashMap instead, but I don't expect any perf improvements.

Reply:
probably, query-operations in generic operation threads are using it, this is outside of partition thread scope and CHM is a must in this case.